src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -53,30 +54,42 @@ class SessionService(
     val requestedBookableStartDate = today.plusDays(noticeDaysMin ?: policyNoticeDaysMin)
     val requestedBookableEndDate = today.plusDays(noticeDaysMax ?: policyNoticeDaysMax)
 
+    // ensure the prisoner - if supplied belongs to the same prison as supplied prisonCode
+    prisonerId?.let {
+      prisonerValidationService.validatePrisonerIsFromPrison(prisonerId, prisonCode)
+    }
+
     var sessionTemplates = sessionTemplateRepository.findValidSessionTemplatesByPrisonCode(
       prisonCode,
       requestedBookableStartDate,
       requestedBookableEndDate
     )
-
     sessionTemplates = filterSessionsTemplatesForLocation(sessionTemplates, prisonerId)
 
     var sessions = sessionTemplates.map {
       buildVisitSessionsUsingTemplate(it, requestedBookableStartDate, requestedBookableEndDate)
     }.flatten()
 
-    if (!prisonerId.isNullOrBlank()) {
```

Comment:
I would revert this back, because the original code needs improving and the code is defensive